### PR TITLE
Dev crysl object

### PR DESCRIPTION
**Pull request by sammekh**
This feature branch includes

- FactoryObject
The factory object gives the option to create either llvm or without llvm objects.
- DefinedObject
This represents the objects without llvm and contains the name and type of the objects.
- ObjectLLVM
This represents the objects with llvm.

Test Cases:
include test cases for five CrySL files, that compare ObJECTS with definedobject names.

### CODE REVIEW

File: CogniCrypt++/CrySL/include/FormatConverter/FactoryObject.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace CCPP {
+class FactoryObject {
+private:
+  std::unordered_map<std::string, std::shared_ptr<Types::Type>> objectTypes;
+  //type(value) will be changed accordingly
+
+public:
+  FactoryObject(CrySLParser::ObjectsContext *);
+  int getObjectsCount();
```
Comment (fabianbs96) [inline CogniCrypt++/CrySL/include/FormatConverter/FactoryObject.h]: Do we actually need this method?

File: CogniCrypt++/CrySL/lib/FormatConverter/FactoryObject.cpp [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  //objectTypes = new std::unordered_map<std::string, std::shared_ptr<Types::Type>>();
+  std::cout << "Debug start in constructor" << std::endl;
+  for(auto objs:oc->objectDecl())
+    objectTypes[objs->Ident()->getText()] = getOrCreateType(objs);
+
+}
+
+int FactoryObject::getObjectsCount()
+{
+  return objectTypes.size();
```
Comment (fabianbs96) [inline CogniCrypt++/CrySL/lib/FormatConverter/FactoryObject.cpp]: Is this correct like this? Probably the number of object-types is lower than the number of objects.

File: CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    std::string objectName;
+    std::shared_ptr<Types::Type> objectType;
+    //type will be changed accordingly
+
+  public:
+    DefinedObject(std::string, std::shared_ptr<Types::Type>);
+    DefinedObject();
+    std::shared_ptr<Types::Type> getType();
+    std::string getName();
+    bool operator==(const DefinedObject &oc);
```
Comment (fabianbs96) [inline CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h]: Make theoperator overload `const`

File: CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  class DefinedObject {
+  private:
+    std::string objectName;
+    std::shared_ptr<Types::Type> objectType;
+    //type will be changed accordingly
+
+  public:
+    DefinedObject(std::string, std::shared_ptr<Types::Type>);
+    DefinedObject();
+    std::shared_ptr<Types::Type> getType();
```
Comment (fabianbs96) [inline CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h]: Make the function `const`

File: CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h (hunk 3 above)
Comment (fabianbs96) [inline CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h]: Make these functions `const`

File: CogniCrypt++/CrySL/include/FormatConverter/ObjectLLVM.h [hunk 5]
```diff
@@ -9,16 +9,16 @@
...
 
   public:
     using TypeT = llvm::Type*;
-    ObjectWithLLVM();
-    ObjectWithLLVM(llvm::Value *);
+    ObjectLLVM();
+    ObjectLLVM(llvm::Value *);
     TypeT getType();
-    bool operator==(const ObjectWithLLVM &oc);
+    bool operator==(const ObjectLLVM &oc);
```
Comment (fabianbs96) [inline CogniCrypt++/CrySL/include/FormatConverter/ObjectLLVM.h]: Make these functions `const`

File: CogniCrypt++/CrySL/include/FormatConverter/FactoryObject.h (hunk 1 above)
Comment (sammekh) [reply CogniCrypt++/CrySL/include/FormatConverter/FactoryObject.h]: This Function was created for unit tests but it's not being used so I've removed it.
Fixed in https://github.com/shariq02/phasar/commit/949851eb07b796f774857fdc0c5e765189cea9e9

File: CogniCrypt++/CrySL/lib/FormatConverter/FactoryObject.cpp (hunk 2 above)
Comment (sammekh) [reply CogniCrypt++/CrySL/lib/FormatConverter/FactoryObject.cpp]: Function Deleted
Fixed in https://github.com/shariq02/phasar/commit/949851eb07b796f774857fdc0c5e765189cea9e9

File: CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h (hunk 3 above)
Comment (sammekh) [reply CogniCrypt++/CrySL/include/FormatConverter/DefinedObject.h]: Fixed in https://github.com/shariq02/phasar/commit/949851eb07b796f774857fdc0c5e765189cea9e9

File: CogniCrypt++/CrySL/include/FormatConverter/ObjectLLVM.h (hunk 5 above)
Comment (sammekh) [reply CogniCrypt++/CrySL/include/FormatConverter/ObjectLLVM.h]: Fixed in https://github.com/shariq02/phasar/commit/949851eb07b796f774857fdc0c5e765189cea9e9

**Comment by sammekh** [comment]
> Please check the code comments and update wherever needed. Refer Doxygen http://www.doxygen.nl/manual/index.html for better code comment standards.

Comments added.
https://github.com/shariq02/phasar/commit/66ae7ddf7c561293a864353aa175c3a1b8f89c63

**Pull request merged**